### PR TITLE
Add hostipc and hostpid to KubernetesPodOperator

### LINKED ISSUE #8402: add HostPID and HostIPC support to KubernetesPodOperator
**Description**

HostPID - Controls whether the pod containers can share the host process ID namespace. Note that when paired with ptrace this can be used to escalate privileges outside of the container (ptrace is forbidden by default).

HostIPC - Controls whether the pod containers can share the host IPC namespace.

**Use case / motivation**
I'd like to be able to start KubernetesPodOperator with HostPID / HostIPC privileges
expose security functionality to the operator.

**Related Issues**

### PR DESCRIPTION
**Pull request by junnplus**
closes: https://github.com/apache/airflow/issues/8402

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by JeffryMAC** [comment]
Thank you @Junnplus !

@kaxil @potiuk

**Comment by mik-laj** [comment]
Hi. We as a community would like to significantly simplify this operator. Does this problem also occur in the new approach? I would rather not add new parameters to this operator, but encourage users to migrate.
https://lists.apache.org/thread.html/rb1bcafea1b8f86e6d155aea7169c05805f7021e90f1636afc8019074%40%3Cdev.airflow.apache.org%3E

File: tests/providers/cncf/kubernetes/operators/test_kubernetes_pod.py [hunk 1]
```diff
@@ -426,3 +426,40 @@ def test_node_selector(self):
         client = ApiClient()
         assert isinstance(result.spec.node_selector, dict)
         assert client.sanitize_for_serialization(result)['spec']['nodeSelector'] == node_selector
+
+    def test_hostpid_and_hostipc(self):
```
Comment (dstandish) [inline tests/providers/cncf/kubernetes/operators/test_kubernetes_pod.py:430]: i think it's better to parameterize a test like this.  will make it easier to see exactly what you're testing, and reduce some duplication also.

**Review by dstandish — CHANGES_REQUESTED** [review]

**Comment by dstandish** [comment]
> Hi. We as a community would like to significantly simplify this operator. Does this problem also occur in the new approach? I would rather not add new parameters to this operator, but encourage users to migrate.
> https://lists.apache.org/thread.html/rb1bcafea1b8f86e6d155aea7169c05805f7021e90f1636afc8019074%40%3Cdev.airflow.apache.org%3E

@mik-laj what's the new approach? I looked at thread but did not see anything replacing this operator...

**Comment by Junnplus** [comment]
@dstandish It seems that KubenertesExecutor is recommended to instead of KubernetesPodOperator.

**Comment by mik-laj** [comment]
@dstandish  it is recommended to use `pod_template_file` or `full_pod_spec` parameter. This way, you have every featture that is available on Kubernetes.

@Junnplus KubernetesPodOperator and KuberntesExecutor are different solutions and both are equally supported.

**Pull request closed without merging**